security/pkg/caclient/config.go

```diff
@@ -23,6 +23,14 @@ type Config struct {
 	// Address of the CA which the CA client calls to
 	CAAddress string
 
+	// Name of the CA provider, e.g Citadel or GoogleCA
+	CAProviderName string
+
+	// IsUsingNewCAProtocol indicates if the node agent is using the old or new CA protocol.
+	// Old protocol: https://github.com/istio/istio/blob/master/security/proto/ca_service.proto
+	// New protocol: https://github.com/istio/istio/blob/master/security/proto/istioca.proto
+	IsUsingNewCAProtocol bool
```

Comment:
Nice documentation. Only concern is the meaning of `New` can be not as straighforward to follow over time.

Reply:
Agreed. But what would you recommend naming them distinguish between the two?

Reply:
UsesIstioCertificateService

Reply:
Or EnableIstioCertificateService. to stay consistent with your env var naming.

Reply:
Sg. Thanks. I will use `IsUsingIstioCertificateService` instead of `IsUsingNewCAProtocol`

Reply:
Oh I see you want it to be a string instead of a bool

Reply:
The bool is fine, as long as the field here is intuitive enough for people to understand it matches which env var we configures from.